compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5406,10 +5406,24 @@ private void checkArrayLibSortFuncArgs(BLangInvocation iExpr) {
             return;
         }
 
-        BLangLambdaFunction keyLambdaFunction = (BLangLambdaFunction) keyFunction;
-        BType returnType = keyLambdaFunction.function.type.getReturnType();
+        DiagnosticPos pos;
+        BType returnType;
+
+        if (keyFunction.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+            pos = keyFunction.pos;
+            returnType = ((BLangSimpleVarRef) keyFunction).type.getReturnType();
+        } else if (keyFunction.getKind() == NodeKind.ARROW_EXPR) {
+            BLangArrowFunction arrowFunction = ((BLangArrowFunction) keyFunction);
+            pos = arrowFunction.params.get(0).pos;
+            returnType = arrowFunction.params.get(0).type;
```

Comment:
Is this correct? Shouldn't we look at the return type?

I guess the validation for ordered type will still work since we use the expected type as the type of the arrow function and invalid types are detected early on.

Reply:
Also, the following seems to result in a bad, sad error at runtime.

```ballerina
import ballerina/lang.array;

public function main() {
    int[] arr = [1, 4, 2];
    int[] sortedArr = arr.sort(array:ASCENDING, (i) => i.toString());
}
```

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/25834